### DISCUSSION TITLE
PageState Helper Methods

### DISCUSSION THREAD

**Discussion by TonyValenti** [oqtane/oqtane.framework#811] [General]
Auditing the Oqtane source code, I see this pattern a lot:
```
 if (PageState.QueryString.ContainsKey("returnurl"))
        {
            _returnUrl = PageState.QueryString["returnurl"];
        }
```

and this pattern:
```
if (PageState.QueryString.ContainsKey("id"))
        {
            _folderId = int.Parse(PageState.QueryString["id"]);
        }
```
First, note that the dictionary is being interrogated twice: once to see if the key exists, and once to fetch the actual value.
Second, note that the second pattern is brittle.  Passing "ASDF" as a query string parameter will crash the app.

Retrieving values from the query string is a common scenario that should be supported in a more robust scenario.

@sbwalker -
Would you be open to a PR that creates extension methods that:
1.  ```TryGetValue``` the value from the dictionary (more efficient than the double lookup)
2.  Optionally ```TryParse``` the value.

I envision the methods being used like any of the following methods:
```
var ID = PageState.QueryString.TryGetInt("ID");
if(PageState.QueryString.TryGetInt("ID", out var ID)){
   ...
}

var Name = PageState.QueryString.TryGetValue("Unknown", "Default Value");

```

**Comment by sbwalker** [comment]
If you look in Oqtane.Shared.Utilities you will notice that these methods already exist - however no one has made the changes to actually implement them throughout the framework.
